presto-spark-base/src/main/java/com/facebook/presto/spark/execution/BatchTaskUpdateRequest.java

```diff
@@ -26,14 +26,17 @@ public class BatchTaskUpdateRequest
 {
     private final TaskUpdateRequest taskUpdateRequest;
     private final Optional<String> shuffleWriteInfo;
+    private final Optional<String> broadcastBasePath;
 
     @JsonCreator
     public BatchTaskUpdateRequest(
             @JsonProperty("taskUpdateRequest") TaskUpdateRequest taskUpdateRequest,
-            @JsonProperty("shuffleWriteInfo") Optional<String> shuffleWriteInfo)
+            @JsonProperty("shuffleWriteInfo") Optional<String> shuffleWriteInfo,
+            @JsonProperty("broadcastBasePath") Optional<String> broadcastBasePath)
     {
         this.taskUpdateRequest = requireNonNull(taskUpdateRequest, "taskUpdateRequest is null");
         this.shuffleWriteInfo = requireNonNull(shuffleWriteInfo, "shuffleWriteInfo is null");
+        this.broadcastBasePath = requireNonNull(broadcastBasePath, "broadcastBasePath is null");
```

Comment:
Should we check that only one of broadcastBasePath or shuffleWriteInfo is set? Ideally, we'd use just one property though.

CC: @shrinidhijoshi @vermapratyush

Reply:
I suggested an approach where this field might not be needed. But yes, if have to retain this field then we should do a sanity check

Reply:
Thanks, added check to validate only one is present